lib/common/utils.c

```diff
@@ -171,7 +171,7 @@ pcmk_daemon_user(uid_t *uid, gid_t *gid)
     static uid_t daemon_uid;
     static gid_t daemon_gid;
     static bool found = false;
-    int rc = pcmk_err_generic;
+    int rc = pcmk_ok;
 
```

Comment:
In which cases do we have an initial failure and success later?
In other words why not cache the rc as well and try just once.

Reply:
If we haven't cached a value yet, rc will be set to crm_user_lookup(), so the initialized value is never used in that case. We only cache a value if the result was successful, so in that case pcmk_ok is always appropriate.

Reply:
Yeah I understood how it is working. Was just asking myself if it would make sense to cache failures as well and if there were cases where it would fail on the first try and be successful sometimes later so that caching failures wouldn't be a good idea.

Reply:
Ah, I understand now -- yes I think we should always retry. I think in most cases any problem severe enough to cause user info to be unavailable will result in a crash or fencing very soon :) so there won't be a retry anyway. But if there is something transient like a brief I/O lock-up, retrying will be helpful.